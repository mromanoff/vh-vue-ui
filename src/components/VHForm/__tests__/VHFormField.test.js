import { shallowMount } from '@vue/test-utils';
import VHFormField from '../VHFormField.vue';

describe('VHFormField.vue', () => {
  let wrapper;

  it('renders default VHFormField', () => {
    wrapper = shallowMount(VHFormField);
    expect(wrapper.props().state).toBe(null);
    expect(wrapper.props().message).toBe(null);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders FormField with state and message', () => {
    wrapper = shallowMount(VHFormField, {
      propsData: {
        state: 'test state',
        message: 'this is a test message',
      },
    });
    expect(wrapper.props().state).toBe('test state');
    expect(wrapper.props().message).toBe('this is a test message');
    expect(wrapper).toMatchSnapshot();
  });
});
