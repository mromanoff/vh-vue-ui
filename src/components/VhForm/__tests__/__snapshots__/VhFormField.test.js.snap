// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VhFormField.vue renders FormField with state and message 1`] = `
<div class="Form-field Form-field--medium Form-field--light Form-field--test state">
  <p class="Form-message">
    this is a test message
  </p>
</div>
`;

exports[`VhFormField.vue renders default VhFormField 1`] = `
<div class="Form-field Form-field--medium Form-field--light">
  <p class="Form-message" style="display: none;">

  </p>
</div>
`;
